server.go

```diff
@@ -283,10 +209,47 @@ func Serve(ctx context.Context, c *websocket.Conn, execer Execer, options *Optio
 	}
 }
 
-func sendExitCode(_ context.Context, exitCode int, conn net.Conn) error {
+// withSession wraps the command in a session if screen is available.
+func (srv *Server) withSession(ctx context.Context, id string, command *Command, execer Execer, options *Options) (*Command, error) {
+	// If screen is not installed spawn the command normally.
+	_, err := exec.LookPath("screen")
+	if err != nil {
+		flog.Info("`screen` could not be found; session %s will not persist", id)
+		return command, nil
+	}
+
+	var s *Session
+	srv.sessionsMutex.Lock()
+	if rawSession, ok := srv.sessions.Load(id); ok {
+		if s, ok = rawSession.(*Session); !ok {
+			return nil, xerrors.Errorf("found invalid type in session map for ID %s", id)
+		}
+	} else {
+		s = NewSession(id, command, execer, options)
+		srv.sessions.Store(id, s)
+		go func() { // Remove the session from the map once it closes.
+			defer srv.sessions.Delete(id)
```

Comment:
The delete happens outside the context of holding the mutex, since it is on a different goroutine.

Reply:
True, I was not sure if locking around the delete would change much functionally since `sync.Map` is threadsafe so I left it out.  Vaguely I was thinking if loading and deleting happens at the same time whether it blocks on the mutex we add or the mutex internal to `sync.Map` it would be more or less equivalent.

I think there might be a gap in my logic though.  And it does make sense to be explicit even if not.